pkg/instancetype/controller/vm/controller_test.go

```diff
@@ -487,32 +488,29 @@ var _ = Describe("Instance type and Preference VirtualMachine Controller", func(
 				Name: instancetypeObj.Name,
 				Kind: "foobar",
 			}
-
-			_, err := instancetypeController.Sync(vm, vmi)
-			Expect(err).To(MatchError(ContainSubstring("got unexpected kind in InstancetypeMatcher")))
-			testutils.ExpectEvents(recorder, common.FailedCreateVirtualMachineReason)
+			syncVM, err := instancetypeController.Sync(vm, vmi)
+			Expect(err).To(MatchError(ContainSubstring("failure to find instance type or preference resources referenced by the VirtualMachine")))
+			Expect(controllerpkg.NewVirtualMachineConditionManager().HasCondition(syncVM, virtv1.VirtualMachineInstancetypeNotFound)).To(BeTrue())
```

Comment:
Wouldn't be better to extract the `controllerpkg.NewVirtualMachineConditionManager()` as variable, since you are using it on multiple places?

Reply:
I've switched over to using the HasConditionTrue matcher instead here.